stake/src/component.rs

```diff
@@ -172,6 +172,21 @@ impl Staking {
                 // Inform tendermint that the validator is no longer active.
                 self.tm_validator_updates.insert(identity_key.clone(), 0);
 
+                if new_state == Inactive {
+                    // Start tracking the validator's uptime with a new uptime tracker.
+                    // This overwrites any existing uptime tracking, regardless of whether
+                    // the validator was recently in the active set.
+                    self.state
+                        .set_validator_uptime(
+                            identity_key,
+                            Uptime::new(
+                                self.state.get_block_height().await?,
+                                self.state.signed_blocks_window_len().await? as usize,
+                            ),
+                        )
+                        .await;
+                }
+
```

Comment:
Do we need to write this in, specially, only when the validator becomes `Inactive`, rather than just updating the gauge? We'll clear the uptime tracker if it ever becomes active, and we'll never read it while it's non-Active.

Reply:
It doesn't seem strictly necessary but I thought it was what you were suggesting in your comment:

> We could also wipe the bitflag when the validator becomes inactive, in order to get this behavior?